streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -269,111 +263,112 @@ public void testLocalThreadCloseWithoutConnectingToBroker() {
         props.setProperty(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
         props.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, NUM_THREADS);
 
-        final StreamsBuilder builder = new StreamsBuilder();
         // make sure we have the global state thread running too
         builder.table("anyTopic");
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
-        streams.start();
-        streams.close();
+        try {
+            streams.start();
+        } finally {
+            streams.close();
```

Comment:
The fact that we are having to do this in our tests suggests to me that maybe we should think about making KafkaStreams `AutoCloseable`. Should I go ahead and make a jira for this?

Reply:
In general I am in favor to this. \cc @guozhangwang @bbejeck Thoughts?

Reply:
+1 for me on `AutoCloseable`